python-spec/src/somacore/query/query.py

```diff
@@ -305,6 +319,7 @@ def _read(
         *,
         column_names: AxisColumnNames,
         X_layers: Sequence[str],
+        obsm_keys: Sequence[str] = [],  # TODO: Add obsp_keys, varm_keys, varp_keys
     ) -> "_AxisQueryResult":
         """Reads the entire query result into in-memory Arrow tables.
```

Comment:
This routine _used_ to only return Arrow objects.  At some point, it looks like it was modified to return SciPy objects.

(looks like this was from a PR landed by @thetorpedodog )

We should at least clean up the comments so they match the code. I'm not sure if there are other implications, other than this routine being less general (i.e., it is now only useful in the to_anndata code path, whereas it was _in theory_ more general previously)

Reply:
As a second thought:  if this code path is dedicated to `to_anndata`, we should name and document it as such.

I haven't given it a huge amount of thought, but I am not opposed to this in principle.  In theory it is a private API, which we can mod at will

Reply:
I changed the docs and remove the part where it mentions Arrow objects. We can take one step further and make the class 100% specific to `to_anndata`.